From eb20785b4cdbfd170ec9c4fdb6a5f114fbd16804 Mon Sep 17 00:00:00 2001
From: Venkatesh Viswanathan <venkateshv@ti.com>
Date: Mon, 28 Mar 2011 19:48:34 +0530
Subject: OMAP4: V4L2: Correction for q-dq sequence

Black frames seen at start of VF preview & av playback.
1.DQ failure happens in overlay resulting in stream off.
This is manifesting as a blank screen as overlay is deleted.
Poll function is used to timeout on any event from kernel
and reclaim the buffers.after analysis of the Q and DQ sequence,
Q of 0,1,2,3,4 happens but DQ of 0 happens but DQ of 2 never
happens.

Change-Id: Ife35cae0a80e839da2c2a1b815de415b0741a9da
Signed-off-by: Sujeet Baranwal <s-baranwal@ti.com>
Signed-off-by: Venkatesh Viswanathan <venkateshv@ti.com>
---
 drivers/media/video/omap/omap_vout.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/omap/omap_vout.c b/drivers/media/video/omap/omap_vout.c
index 3b33ecb..8727896 100644
--- a/drivers/media/video/omap/omap_vout.c
+++ b/drivers/media/video/omap/omap_vout.c
@@ -2741,6 +2741,7 @@ static int vidioc_qbuf(struct file *file, void *fh,
 
 	/* also process frame if displayed on a manual update screen */
 	if ((vout->wb_enabled || manually_updated(vout)) &&
+			!vout->first_int &&
 			vout->cur_frm &&
 			vout->next_frm &&
 			vout->cur_frm->i == vout->next_frm->i &&
-- 
1.6.0.4

